pmd-java/src/main/java/net/sourceforge/pmd/lang/java/oom/ParameterizedMetricKey.java

```diff
@@ -35,25 +35,13 @@ private ParameterizedMetricKey(MetricKey<? extends Metric> key, MetricVersion ve
 
     @Override
     public String toString() {
-        return "ParameterizedMetricKey{key=" + key + ", version=" + version + '}';
+        return "ParameterizedMetricKey{key=" + key.name() + ", version=" + version.name() + '}';
     }
 
 
     @Override
     public boolean equals(Object o) {
-        if (this == o) {
-            return true;
-        }
-        if (o == null || getClass() != o.getClass()) {
-            return false;
-        }
-
-        ParameterizedMetricKey that = (ParameterizedMetricKey) o;
-
-        if (!key.equals(that.key)) {
-            return false;
-        }
-        return version.equals(that.version);
+        return this == o;
```

Comment:
In the metrics framework not yet, but later on, when we integrate it with the analysis cache. 

Currently "RuleViolations" are serialized into a cache file, and - if the file has not been modified since the last pmd run (and pmd is run with the same rules) - the file is not analyzed again and the RuleViolations from the cache are simply returned.